src/rapid_pro_to_engagement_db/rapid_pro_to_engagement_db.py

```diff
@@ -174,15 +180,34 @@ def sync_rapid_pro_to_engagement_db(rapid_pro, engagement_db, uuid_table, rapid_
     # (If the cache or a contacts file for this workspace don't exist, `contacts` will be `None` for now)
     contacts = _get_contacts_from_cache(cache)
 
-    dataset_to_sync_stats = dict()  # of '{flow_name}.{flow_result_field}' -> RapidProToEngagementDBSyncStats
-    for flow_config in rapid_pro_config.flow_result_configurations:
-        sync_stats = RapidProToEngagementDBSyncStats()
+    # Check the configs are the same before proceeding with cached data
+    cached_flow_result_configs = _get_flow_result_configs_from_cache(cache)
+    if cached_flow_result_configs is None and cache is not None and not dry_run:
+        cache.set_flow_result_configs(rapid_pro_config.flow_result_configurations)
+    else:
+        cached_flow_result_configs = [d.to_dict() for d in cached_flow_result_configs]
+        updated_flow_result_configs = [config for config in rapid_pro_config.flow_result_configurations \
+                                        if config.to_dict() not in cached_flow_result_configs]            
+        if len(updated_flow_result_configs) > 0:
+            seen = set()
+            for config in updated_flow_result_configs:
+                if config.flow_name not in seen:
+                    cache.reset_latest_run_timestamp(rapid_pro.get_flow_id(config.flow_name))
+                    seen.add(config.flow_name)
+            if not dry_run:
+                cache.set_flow_result_configs(rapid_pro_config.flow_result_configurations)
```

Comment:
Can you extract this into a function?

Reply:
extracted into `_update_cache_with_changes_in_flow_result_configs`